### ISSUE TITLE
HTTPS rollout

### ISSUE THREAD

**Issue by alastair** [MTG/freesound#574]
Some notes for things we need to fix before we can use https
- [x] Make captcha on registration page work (use `//` protocol-less address)
- [x] Make nginx take a round-robin server out of rotation if it's not available
- [x] Set http host on round robin
- [x] Do not show images from non-https sites

**Comment by qubodup** [comment]
Once https is done, please also only use https://freesound.org and make https://www.freesound.org, http://freesound.org, http://www.freesound.org use https://freesound.org

**Comment by alastair** [comment]
www -> nowww redirect is https://github.com/MTG/freesound/issues/664


The last thing to be done here is make linked images secure. This is extra work, probably requiring us to find images, download them, and cache them ourselves. Perhaps we should open a new ticket for it.

**Comment by ffont** [comment]
@alastair @bdejong @andrebola  What do you think we should do with this ticket?
I think it would be nice to move completely to https (still allow http in the API though).
We should find a solution for the images problem.

**Comment by bdejong** [comment]
This would be awesome especially if we could always forward ppl to HTTPS...

The image thing, ... Local caching? This could also speed up rendering
pages with external images and remove dependencies on external servers.

**Comment by bdejong** [comment]
Alternative/easier: change the HTML parsing:

If an image URL does not start with https, turn the image link into a URL.
Even easier: disable image linking, but this might make some people angry.

**Comment by alastair** [comment]
I like this option, we should just show images which are on an https link and otherwise show a link. Not sure how much of a change we need to make to our parser. I don't want to start the discussion about changing our comments and descriptions to Markdown, but perhaps we should put it on the list.... ;)

**Comment by bdejong** [comment]
@alastair using markdown would not solve this particular problem though ;-)

More importantly: keep in mind that all HTML is valid markdown... So by switching to markdown you actually open up to even more chaos ;-)

**Comment by qubodup** [comment]
Hi,

> disable image linking

Totally OK for comments, personal messages and descriptions. A bit uncool for forum posts.

> just show images which are on an https link and otherwise show a link

Sounds totally OK. Ideally paired with a notification to the poster "Freesound only supports embedded images from https:// domains. The image embed code was replaced with a link"

**Comment by alastair** [comment]
I think we're agreed that images are an integral part of the freesound experience, so I don't think we want to remove them.

We'll work towards filtering out non-https images, with a warning message.

**Comment by alastair** [comment]
@ffont suggested that we implement this by using a template tag, so we change `<img src="http://...>` to `<a href="">` at display time, instead of rewriting the user's text.
If we see an http image url we can show a message to the user while they are writing the comment to say that this will happen.

**Comment by alastair** [comment]
https://nickcraver.com/blog/2017/05/22/https-on-stack-overflow/ is an interesting description of the problems that stackoverflow had switching to https

**Comment by alastair** [comment]
HTTPs not fully rolled out

**Comment by alastair** [comment]
I had a look at the statistics in google analytics. In August 2017 we had about 10 sessions (out of 1.2 million) which used Windows xp / internet explorer 6. There are also a handful of old android and firefox versions. These aren't supported with https, so we'll lose these users. We spoke about this in chat and decided that it's an OK tradeoff to not support these users. We'll go ahead with the automatic switch to https

**Comment by alastair** [comment]
We released HTTPS redirects. 

The main freesound site redirects from http->https
If the user visits with www, we redirect directly to https (instead of first to nowww, then to https)
API requests are not redirected, except for `/apiv2/apply`

I was wrong with my comment at https://github.com/MTG/freesound/pull/982#discussion-diff-148473244R95, we have to redirect to https in django because we have no way of knowing in haproxy if the user is in a browser or with an API client
